File: PYTHON/B2476.py
def calculate_prize(rd):
    rd_set = set(rd)
    
    if len(rd_set) == 1:  
        return 10000 + (rd[0] * 1000)
    elif len(rd_set) == 2:  
        for num in rd_set:
            if rd.count(num) == 2:
                return 1000 + (num * 100)
    else:  
        return max(rd) * 100

def main():
    N = int(input())
    
    max_prize = 0
    
    for _ in range(N):
        rd = list(map(int, input().split()))
        prize = calculate_prize(rd)
        max_prize = max(max_prize, prize)
    
    print(max_prize)

if __name__ == "__main__":
    main()

File: C/B2476.c
#include <stdio.h>

int main() {
    int N;
    scanf("%d", &N);

    int maxPrize = 0;

    for (int i = 0; i < N; i++) {
        int rd[3];
        scanf("%d %d %d", &rd[0], &rd[1], &rd[2]);

        int prize = 0;

        if (rd[0] == rd[1] && rd[1] == rd[2]) {
            prize = 10000 + rd[0] * 1000;
        }

        else if (rd[0] == rd[1] || rd[1] == rd[2] || rd[0] == rd[2]) {
            prize = 1000 + rd[1] * 100;
        }

        else {
            int maxrd = (rd[0] > rd[1]) ? ((rd[0] > rd[2]) ? rd[0] : rd[2]) : ((rd[1] > rd[2]) ? rd[1] : rd[2]);
            prize = maxrd * 100;
        }

        if (prize > maxPrize) {
            maxPrize = prize;
        }
    }

    printf("%d\n", maxPrize);

    return 0;
}
